scripts/test.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# This script is executed by Travis CI for every successful push (on any branch, PR or not).
+# It runs some basic tests on pages. If the build is also a PR, additional
+# checks are run through the check-pr script, and any message or error is sent
+# to tldr-bot to be commented on the PR.
+#
+# NOTE: must be run from the repository root directory to correctly work!
+# NOTE: `set -e` is applied conditionally only if needed.
+
+# Default test.
+function default_test {
+  markdownlint pages*/**/*.md
+  tldr-lint ./pages
+}
+
+# Special test for Travis CI pull request builds.
+# Runs default_test collecting errors for tldr-bot.
+function pr_test {
+  errs=`default_test 2>&1`
+
+  if [ -n "$errs" ]; then
+    echo -e "Test failed!\n$errs\n" >&2
+    echo 'Sending errors to tldr-bot.' >&2
```

Comment:
Refactor these 2 lines in a function and call them. You are using them in pr_test and pr_check.

Reply:
I'm not sure if it makes sense to refactor this in a function. It would add a lot of boilerplate code just for a simple command. The two messages are different and the two invocations of the python script are different too, this would mean creating a function that would need to be called with two arguments and have additional logic inside, it would make the code harder to read IMHO.

Reply:
>  this would mean creating a function that would need to be called with two arguments and have additional logic inside,

Yep, there would be 2 arguments, but I fail to see where you would need additional logic ? It will just be `echo -n "$msg" | python3 scripts/send_to_bot.py $command` right ?

Reply:
Well, to refactor L24 and L25 into a function they should be the same as L37 and L38. In particular L24 and L37 differ, one says "sending errors" and one says "sending check results". 

To refactor both L24 and L25 into a function would mean adding additional logic in the function. Otherwise, refactoring only L25 basically means creating a function to substitute one single line of code... which seems like over-engineering the thing IMHO.

Reply:
Oh, my bad, I missed that the log messages were different. You can parameterize that too.

But if you feel that's complicating things too much, let's use better names like I mentioned [here](https://github.com/tldr-pages/tldr/pull/3576/files#r347075482).

Reply:
I think it's probably fine, but we could have a function like this:

```bash
make_comment() {
    type="${1}" # either "error" or "check"
    message = "${2}"
    echo -n "${message}" | python3 scripts/send_to_bot.py "${type}"
}
```

The 2nd param is optional - we could leave it as taking stdin I guess.

Reply:
This is basically refactoring one line. Which is what @mebeim mentioned here
> Otherwise, refactoring only L25 basically means creating a function to substitute one single line of code... which seems like over-engineering the thing IMHO.

Reply:
All I'm saying is that moving that command into a function would mean writing something like this:

```bash
function send_to_bot {
	echo -n "$1" | python3 scripts/send_to_bot.py $0
}

# ...

send_to_bot error "$errs"
```

instead of just:

```bash
echo -n "$errs" | python3 scripts/send_to_bot.py error
```

As I said it looks like over-engineering the thing, so I would not advise this unless you guys have a concrete example that makes more sense.

Reply:
I am with you @mebeim on this one :) . I was just explaining what you already mentioned to @sbrl.

Reply:
OK, I should have addressed all remaining comments, leaving this open for @sbrl to read.